datastore/blob/azure_storage_api.py

```diff
@@ -0,0 +1,266 @@
+    GetFolderUUIDError,
+    FolderListError,
+    GenerateHashError,
+    CreateDirectoryError,
+)
+
+"""
+---- user-container based structure -----
+- container name is user id
+- whenever a new user is created, a new container is created with the user uuid
+- inside the container, there are project folders (project name = project uuid)
+- for each project folder, there is a json file with the project info and creation
+date, in the container
+- inside the project folder, there is an image file and a json file with
+the image inference results
+"""
+
+
+async def generate_hash(image):
+    """
+    generates a hash value for the image to be used as the image name in the container
+    """
+    try:
+        hash = hashlib.sha256(image).hexdigest()
+        return hash
+
+    except GenerateHashError as error:
+        print(error)
+
+
+async def mount_container(connection_string, container_uuid, create_container=True,tier="user"):
+    """
+    Creates a container_client as an object that can be used in other functions.
+    
+    Parameters:
+    - connection_string: the connection string to the azure storage account
+    - container_uuid: the uuid of the container (usually the user uuid)
+    - create_container: a boolean value to specify if the container should be created if it doesnt exist (default is True)
+    - tier: the tier of the container (default is user, should be changed if the structure changes to accomodate other type of containers)
+    
+    Returns:
+    - container_client: the container client object
+    """
+    try:
+        blob_service_client = BlobServiceClient.from_connection_string(
+            connection_string
+        )
+        if blob_service_client:
+            container_name = "{}-{}".format(tier,container_uuid)
+            container_client = blob_service_client.get_container_client(container_name)
+            if container_client.exists():
+                return container_client
+            elif create_container and not container_client.exists():
+                container_client = blob_service_client.create_container(container_name)
+                # create general directory for new user container
+                response = await create_folder(container_client, "General")
+                if response:
+                    return container_client
+                else:
+                    return False
+        else:
+            raise ConnectionStringError("Invalid connection string")
+
+    except MountContainerError as error:
+        print(error)
+        return False
+
+
+async def get_blob(container_client, blob_name):
+    """
+    gets the contents of a specified blob in the user's container
+    """
+    try:
+        blob_client = container_client.get_blob_client(blob_name)
+        blob = blob_client.download_blob()
+        blob_content = blob.readall()
+        return blob_content
+
+    except GetBlobError as error:
+        print(error)
+        return False
```

Comment:
why silently fail instead of throwing an exception?

Reply:
idk, it was imported from the main branch of nachet-backend. I've applied changes here

Reply:
That's what I was referring at yesterday when I said that this file add problem...

I open an issue in the backend related to this: 

* https://github.com/ai-cfia/nachet-backend/issues/61